### ISSUE TITLE
proper JSON representation for Recipe

### ISSUE THREAD

**Issue by mmandlis** [PolymerLabs/arcs#2552]
Currently suggestion.ts declares a `Plan` class (https://github.com/PolymerLabs/arcs/blob/master/src/runtime/plan/suggestion.ts#L34) that is a JSON representation for a `Recipe`, used for de/serialization to/from storage and some basic reasoning about prior to plan instantiation, for example performing `RecipeUtil.matchesRecipe` on the JSON form of the plan (now fails on #2550, also `handleConnection` requires `handle` information in some cases).

**Comment by piotrswigon** [comment]
Can you explain why we cannot perform proper plan de-serialization in Recipe instance straight away? Given that we do it before plan instantiation. Is there some gotcha there?

**Comment by mmandlis** [comment]
The original reason to introduce the current JSON format was to make `fromLiteral` method synchronous and not dependent on anything other than the serialized data.
`Manifest.parse` method is async and requires loader and context. Occasionally a RecipeResolver is needed to resolve the deserialized recipe.

I talked to @shans about this and we decided to not implement this issue. 

Instead I will:
- comment on the existing `Plan` class that it is temporary
- update fromLiteral to allow relying on user environment data (but not specific arc though), add `loader` and `context` to Runtime class and pass it to fromLiteral
- RecipeResolver: I need to investigate whether this is still happening and why. If it is not easy to fix, I'll move resolving the recipe to the instantiate method

I will close this issue and file a separate one for this work.
